server/index.js

```diff
@@ -1,32 +1,65 @@
+const cors = require("cors");
+const express = require("express");
+const mysql = require("mysql2")
 
+const app = express();
+
+app.use(express.json())
 app.use(cors())
 
-app.get('/cars', function (req, res) {
-    res.json(cars);
-});
-
-app.listen(4000, function () {
-    console.log('Example app listening on port 4000!');
-});
-
-
-
-const cars = [
-    {made: "KIA", model:"Rio", year:"2017", price: "9600", transmission:"FWD",
-        gearbox:"mechanic", milage:"2281337 km", bodytpe:"Sedan", enginetpe:"Gasoline"},
-    {made: "Hummer", model:"H1", year:"1983",price: "19400", transmission:"AWD",
-        gearbox:"mechanic", milage:"14881337 km", bodytpe:"SUV", enginetpe:"Gasoline"},
-    {made: "Ford", model:"Raptor", year:"2017",price: "41200", transmission:"AWD",
-        gearbox:"automatic", milage:"228777 km", bodytpe:"Truck", enginetpe:"Diesel"},
-    {made: "Ford", model:"Interceptor", year:"2014",price: "11800", transmission:"RWD",
-        gearbox:"automatic", milage:"239000 km", bodytpe:"Sedan", enginetpe:"Gasoline"},
-    {made: "Audi", model: "RS6", year: "2011", price: "32000", transmission: "4WD",
-        gearbox: "automatic", milage: "44000", bodytpe: "Wagon", enginetpe: "Gasoline"},
-    {made: "Toyota", model: "Prius", year: "2007", price: "13000", transmission: "FWD",
-        gearbox: "automatic", milage: "470000", bodytpe: "Hatchback", enginetpe: "Hybrid"},
-    {made: "Audi", model: "E-tron", year: "2021", price: "142000", transmission: "4WD",
-        gearbox: "automatic", milage: "41200", bodytpe: "SUV", enginetpe: "Electric"}
-]
\ No newline at end of file
+app.listen(process.env.PORT || 4000, () => {
+    console.log('Server is running')
+})
+
+const db = mysql.createPool({
+    host:"eu-cdbr-west-03.cleardb.net",
+    user:"b26e27e26d1df4",
+    password:"e77f53d7",
+    database:"heroku_147c2869609d665"
+})
+
+db.query('select 1 + 1', (err, rows) => { /* */ });
+
+app.get('/', (req,res)=>{
+    res.json("This is server side")
+})
+
+app.get('/made', (req,res)=>{
+    const made = "SELECT * FROM carbrands"
+    db.query(made,(err,data)=>{
+        if(err) return res.json(err)
+        return res.json(data)
+    })
+})
+
+app.get('/bodytypes', (req,res)=>{
+    const made = "SELECT * FROM bodytypes"
+    db.query(made,(err,data)=>{
+        if(err) return res.json(err)
+        return res.json(data)
+    })
+})
+
+app.get('/engines', (req,res)=>{
+    const made = "SELECT * FROM enginetypes"
+    db.query(made,(err,data)=>{
+        if(err) return res.json(err)
+        return res.json(data)
+    })
+})
+
+app.get('/gearboxes', (req,res)=>{
+    const made = "SELECT * FROM gearboxes"
+    db.query(made,(err,data)=>{
+        if(err) return res.json(err)
+        return res.json(data)
+    })
+})
+
+app.get('/transmissions', (req,res)=>{
+    const made = "SELECT * FROM transmissions"
+    db.query(made,(err,data)=>{
+        if(err) return res.json(err)
+        return res.json(data)
+    })
+})
```

Comment:
in IndexJs file only main data about run server, cors, should be placed. Create separate files with calling apis from there.
And I don't see typescript